module/mempool/stdmap/backend.go

```diff
@@ -101,18 +101,18 @@ func (b *Backdata) Hash() flow.Identifier {
 type Backend struct {
 	sync.RWMutex
 	Backdata
-	limit            uint
-	eject            EjectFunc
-	ejectionCallback OnEjection
+	limit             uint
+	eject             EjectFunc
+	ejectionCallbacks []mempool.OnEjection
```

Comment:
@zhangchiqing I went back to using a list of ejection callbacks. After trying around a bit, I found this solution to be much more general compared to a single callback. Trying to give a high-level summary here:
* some mempools don't store the entities directly in the backend, but use composite structure, e.g.: https://github.com/onflow/flow-go/blob/7c481533a9d41d4c435d2dbb9542e94452c07928/module/mempool/stdmap/incorporated_results.go#L45
* To not expose this internal details to an externally-provided ejection callback, the callback needs to be wrapped before it is injected into the `Backend`. 
* By allowing the backend to serve multiple callbacks:
   - it is trivial for a mempool to wrap any externally-provided callback with the necessary logic before registering the (wrapped) ejection callback with the `Backend`
   - In contrast, when the `Backend` only accepts a single callback, the situation is much more complex for a mempool that has an internal ejection callback as well as accepts external ejection callbacks (like [`IncorporatedResults`](https://github.com/onflow/flow-go/blob/7c481533a9d41d4c435d2dbb9542e94452c07928/module/mempool/stdmap/incorporated_results.go)):
    - the mempool must implement a multiplexer which serves its internal mempool as well as an external mempool

To me, it seemed much more straightforward, to include the functionality to serve multiple callbacks into the Backend, specifically so as it is only adds two lines of code here. If you have other ideas, lets sync Leo. I tried for quite some time and this seemed to be by far the easiest (and least error-prone) solution.